packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -875,6 +852,65 @@ export const uploadTemplateToS3 = async (
   }
 };
 
+/**
+ Create an event map which will be used to map each incoming CloudFormation event into a category.
+ This is done so as to group events under progress bars for the root project as well as for each category.
+ */
+const createEventMap = (
+  context: $TSContext,
+  projectDetails: $TSObject,
+  resourcesToBeCreated: $TSAny,
+  resourcesToBeUpdated: $TSAny,
+): any => {
+  let eventMap : any = {};
+
+  const { envName } = context.amplify.getEnvInfo();
+  const { projectName } = context.amplify.getProjectConfig();
+  const rootStackName = projectDetails.teamProviderInfo[envName].awscloudformation.StackName;
+
+  // Setting up initial configurations.
+  eventMap.rootStackName = rootStackName;
+  eventMap.envName = envName;
+  eventMap.projectName = projectName;
+  eventMap.rootResources = [];
+  eventMap.eventToCategories = new Map();
+
+  // Type script throws an error unless I explicitly convert to string
+  const resourcesUpdated = getAllUniqueCategories(resourcesToBeUpdated).map(item => `${item}`);
+  const resourcesCreated = getAllUniqueCategories(resourcesToBeCreated).map(item => `${item}`);
+
+  let categories = Object.keys(projectDetails.amplifyMeta);
+  categories = categories.filter(category => category !== 'providers');
+  eventMap.categories = [];
+  categories.forEach(category => {
+    const resources = Object.keys(projectDetails.amplifyMeta[category]);
+    resources.forEach(resource => {
+      const resourceKey = category + resource;
+      // root resource key is made this way
+      eventMap.rootResources.push({ category: `${category}-${resource}`, key: resourceKey });
+
+      const fileSystem = context.filesystem;
+      // Getting corresponding cfn template files
+      const { resourceDir, cfnFiles } = getCfnFiles(category, resource);
+      cfnFiles.forEach(file => {
```

Comment:
this is way too much nesting. I would break this function out into it's own named function and also break out the `resources.forEach()` on line 887 into it's own function as well